File: Framework/CSharp/Framework/Framework/SmartEnum.cs
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
using System;
using System.Collections.Generic;
using System.Linq;

namespace Smartkernel.Framework
{
	/// <summary>
	/// 枚举辅助操作类
	/// </summary>
	public static class SmartEnum
	{
		/// <summary>
		/// 将字符串表示形式的枚举解析成枚举类型
		/// </summary>
		/// <typeparam name="T">枚举类型</typeparam>
		/// <param name="value">字符串表示形式</param>
		/// <returns>解析的结果</returns>
		public static T Parse<T>(string value)
		{
			var type = typeof(T);
			return (T)Enum.Parse(type, value, true);
		}

		/// <summary>
		/// 获得某种枚举类型下的所有枚举值
		/// </summary>
		/// <typeparam name="T">枚举类型</typeparam>
		/// <returns>所有值</returns>
		public static List<T> GetAll<T>()
		{
			var enumType = typeof(T);

			var values = new List<T>();

			var fields = from field in enumType.GetFields()
			                      where field.IsLiteral
			                      select field;

			foreach (var field in fields)
			{
				var value = field.GetValue(enumType);
				values.Add((T)value);
			}
			return values.ToList();
		}
	}
}

File: Framework/Java/framework/framework/smartkernel/framework/SmartEnum.java
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
package smartkernel.framework;

import java.util.*;

/**
 * 枚举辅助操作类
 * 
 */
public class SmartEnum {
	/**
	 * 将字符串表示形式的枚举解析成枚举类型
	 * 
	 * @param type
	 *            ，枚举类型
	 * @param value
	 *            ，字符串表示形式
	 * @return，解析的结果
	 */
	public static <T extends Enum<T>> T parse(Class<T> type, String value) {
		ArrayList<T> list = SmartEnum.getAll(type);
		for (T one : list) {
			if (one.name().equalsIgnoreCase(value)) {
				return (T) one;
			}
		}
		throw new RuntimeException("枚举值不存在！");
	}

	/**
	 * 获得某种枚举类型下的所有枚举值
	 * 
	 * @param type
	 *            ，枚举类型
	 * @return，所有值
	 */
	public static <T extends Enum<T>> ArrayList<T> getAll(Class<T> type) {
		ArrayList<T> values = new ArrayList<T>();

		Iterator<T> iterator = EnumSet.allOf(type).iterator();
		while (iterator.hasNext()) {
			String currentString = iterator.next().name();
			values.add((T) Enum.valueOf(type, currentString));
		}
		return values;
	}
}
